Thank you for the survey report.


I remembered a cmake warning log in actions build.
```
[ 99%] Building CXX object CMakeFiles/flycast.dir/core/windows/winmain.cpp.obj
In file included from D:\a\flycast\flycast\core\windows\winmain.cpp:22:
C:/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/x86_64-w64-mingw32/include/winsock2.h:15:2: warning: #warning Please include winsock2.h before windows.h [-Wcpp]
 #warning Please include winsock2.h before windows.h
  ^~~~~~~
[100%] Building CXX object CMakeFiles/flycast.dir/core/windows/win_vmem.cpp.obj
```

--- a/core/gdxsv/gdxsv.cpp
+++ b/core/gdxsv/gdxsv.cpp
@@ -44,21 +44,11 @@ void Gdxsv::Reset() {
         enabled = false;
         return;
     }
+    enabled = true;
 
-#ifdef _WIN32
-    if (enabled) {
-        WSACleanup();
-    }
-    WSADATA wsaData;
-    if (WSAStartup(MAKEWORD(2, 0), &wsaData) != 0) {
-        ERROR_LOG(COMMON, "WSAStartup failed. errno=%d", get_last_error());
-        return;
-    }
-#else
+#ifdef __APPLE__
     signal(SIGPIPE, SIG_IGN);
 #endif
-    
-    enabled = true;
 
     if (!net_thread.joinable()) {
         NOTICE_LOG(COMMON, "start net thread");

--- a/core/windows/winmain.cpp
+++ b/core/windows/winmain.cpp
@@ -431,6 +431,11 @@ void os_CreateWindow()
 	CreateMainWindow();
 	InitRenderApi();
 #endif	// !USE_SDL
+	WSADATA wsaData;
+	if (WSAStartup(MAKEWORD(2, 0), &wsaData) != 0) {
+		ERROR_LOG(COMMON, "WSAStartup failed. errno=%d", get_last_error());
+		return;
+	}
 }
 
 void DestroyMainWindow()
@@ -446,6 +451,7 @@ void DestroyMainWindow()
 		DestroyWindow(hWnd);
 		hWnd = NULL;
 	}
+	WSACleanup();
 }
 
 void* libPvr_GetRenderTarget()

--- a/core/windows/winmain.cpp
+++ b/core/windows/winmain.cpp
@@ -433,7 +433,7 @@ void os_CreateWindow()
 #endif	// !USE_SDL
 	WSADATA wsaData;
 	if (WSAStartup(MAKEWORD(2, 0), &wsaData) != 0) {
-		ERROR_LOG(COMMON, "WSAStartup failed. errno=%d", get_last_error());
+		ERROR_LOG(COMMON, "WSAStartup failed. errno=%d", WSAGetLastError());
 		return;
 	}
 }

--- a/core/windows/winmain.cpp
+++ b/core/windows/winmain.cpp
@@ -1,4 +1,5 @@
-#define _WINSOCKAPI_    // stops windows.h including winsock.h
+#include <winsock2.h>
+
 #include "oslib/oslib.h"
 #include "oslib/audiostream.h"
 #include "imgread/common.h"
@@ -19,7 +20,6 @@
 #include "oslib/host_context.h"
 #include "../shell/windows/resource.h"
 
-#include <winsock2.h>
 #include <ws2ipdef.h>
 #include <iphlpapi.h>
 #include <netioapi.h>

